src/passes/Inlining.cpp

```diff
@@ -199,11 +235,55 @@ struct Inlining : public Pass {
     // return whether we did any work
     return inlined.size() > 0;
   }
+
+  bool canInline(int numUses) {
+    return numUses == SINGLE_USE;
+  }
+
+  bool worthInlining(Function* func) {
+    return Measurer::measure(func->body) <= INLINING_SIZE_LIMIT;
```

Comment:
Yeah, this is something LLVM LTO should be able to do. But we can do it much faster ;)